sshd-core/src/test/java/org/apache/sshd/client/ClientTest.java

```diff
@@ -1530,6 +1532,29 @@ public void testConnectUsingIPv6Address() throws IOException {
         }
     }
 
+    @Test // see SSHD-1295
+    public void testConnectTimeout() throws Exception {
+        client.start();
+        try {
+            ConnectFuture future = client.connect(getCurrentTestName(), TEST_LOCALHOST, port);
+            try {
+                future.verify(1);
+                fail("Timeout expected");
```

Comment:
Somehow this test now fails in the CI tests here. See https://github.com/apache/mina-sshd/runs/8283257340?check_suite_focus=true . Any idea why?

Reply:
See below.